### PR TITLE
GetRoleSalary implemented for agents to read salaries from rules

### PR DESCRIPTION
**Pull request by 777jameskim**
# Summary

Requires PR279 to go through

## Additional Information

Auxiliary information goes here.

## Test Plan

- Add information on how you tested your changes.

### CODE REVIEW

**Comment by 777jameskim** [comment]
@Spruudis i @ u

File: internal/common/baseclient/baseSpeaker.go [hunk 1]
```diff
@@ -11,11 +12,17 @@ type BaseSpeaker struct {
...
-		JudgeSalary: 10,
+		JudgeSalary: s.GetJudgeSalary(),
 		ActionTaken: true,
 	}
 }
 
+// GetJudgeSalary gets the Judge's salary from the rules.
+func (s *BaseSpeaker) GetJudgeSalary() shared.Resources {
+	JudgeSalaryRule := rules.AvailableRules["salary_cycle_judge"]
+	return shared.Resources(JudgeSalaryRule.ApplicableMatrix.At(0, 1))
```
Comment (Spruudis) [inline internal/common/baseclient/baseSpeaker.go]: i think if this is to be a utility function it should check rulesinplay and return a struct of something like {value int, rulepresent bool}

File: internal/common/roles/speaker.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import "github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 // optionally implemented by individual islands.
 type Speaker interface {
 	PayJudge() shared.SpeakerReturnContent
+	GetJudgeSalary() shared.Resources
```
Comment (Spruudis) [inline internal/common/roles/speaker.go]: Does this seem like it should be a utility function in rules more than it should be a function they have to override? We should not change the interface I feel

**Review by Spruudis — CHANGES_REQUESTED** [review]

File: internal/common/roles/speaker.go (hunk 2 above)
Comment (777jameskim) [reply internal/common/roles/speaker.go]: Yeah that's just cuz i made the function first and later realised that's where it belongs 😛 
I can merge the second function into the first, gimme a sec

**Review by Spruudis — APPROVED** [review]
Looking good and no I'm not talking about the code

**Pull request merged**